### DISCUSSION TITLE
Getting trace level log for geckodriver

### DISCUSSION THREAD

**Discussion by ltanase77** [php-webdriver/php-webdriver#1046] [Q&A]
Hi,
I am using php-webdriver (version 1.13) in a Laravel 8 project and I have an error: `Failed to read marionette port`. The error appears when WebDriver needs to start the GeckoDriver with this line of code: `$driver = Firefox\FirefoxDriver::start($desiredCapabilities)`; If I start manually firefox and gecko driver and I am using `$driver = RemoteWebDriver::create($serverUrl, $desiredCapabilities);` everything is fine. But this implies that firefox and geckodriver to run permanently.

I have opened a issue with geckodriver team on their github repository, but they are asking me to provide them a trace level log of geckodriver. I have asked to tell me the location where geckodriver save its log files, but they told me that it depends on the Selenium bindings, and to check the support page of php-webdriver. 
In my laravel application I can see only the laravel.log file which contain a stack trace but it does not seem to contain the trace level log for geckodriver. 

Can anyone help me with location of geckodriver log files or how I may obtain them.

**Comment by andrewnicols** [comment]
On mobile but I’ll try and remember to follow up tomorrow with more info,
but you need to specify the log level and items you want to log in your
desired capabilities.

Here’s how I do so in my config generator:

https://github.com/andrewnicols/moodle-browser-config/blob/0464ab33eed6b1f3aa247f3ed849e27493ae2a72/src/ProfileManager.php#L648

This is noted in
https://firefox-source-docs.mozilla.org/testing/geckodriver/TraceLogs.html

**Comment by ltanase77** [comment]
I have tried $firefoxOptions->setOption("log", ["level" => "trace"]) but still do not know where look for geckodriver log file. After I added this line the error was changed from `Failed to read marionette port` to `
```
Facebook\WebDriver\Exception\WebDriverCurlException Curl error thrown for http POST to /session with params: {"capabilities":{"firstMatch":[{"browserName":"firefox","moz:firefoxOptions":{"binary":"/usr/bin/firefox","log":{"level":"trace"},"args":["-headless"],"prefs":{"reader.parse-on-load.enabled":false,"devtools.jsonview.enabled":false}},"acceptInsecureCerts":false}]}}

Operation timed out after 180001 milliseconds with 0 bytes received
```
`

**Comment by ltanase77** [comment]
If I try `$firefoxOptions->addArguments(['-headless', 'log', '["level" => "trace"]'])`  I receive the initial error `Failed to read marionette port`
